doc/protocols.rst

```diff
@@ -0,0 +1,124 @@
+.. py:currentmodule:: lsst.ts.ess
+
+.. _lsst.ts.ess-protocols:
+
+#######################
+Communication Protocols
+#######################
+
+This page describes the communication protocols used by the `EssCSC` to communicate with the ``ts_envsensors`` code running on the Raspberry Pi 4's.
+All communication is done via JSON encoded strings.
+All JSON strings have a common header, indicated by the ``msg_type`` keyword.
+The ``msg_type`` keyword can have the values
+
+- command, indicating a command
+- response, indicating a response to a command
+- telemetry, indicating telemetry sent by the Raspberry Pi 4's
+
+This keyword is needed to be able to distinguish between responses and telemetry in a fail safe manner.
+
+Commands
+--------
+
+Apart from the ``msg_type`` keyword with the ``command`` value, all commands have the following keywords:
+
+* cmd_name: the name of the command
+* cmd_id: a unique, increasing long value determined by the CSC.
+  This is the UNIX timestamp in milliseconds when the command was sent
+* parameters: a set of name/value pairs where each name represents the name of a parameter and each value the value of the parameter
```

Comment:
each value the value of the parameter --> each value of the parameter

Reply:
Actually the sentence is correct but I agree that it is confusing. I rewrote it to `and each value represents the value of the parameter`.